### PR TITLE
Dont set default distro when OSType is Windows

### LINKED ISSUE #2619: Remove or change distro when OS type is Windows
Right now, `"distro"` defaults to `Ubuntu` , even when the OS is Windows. The value is not used for Windows but nevertheless we should remove it for Windows as it may cause confusion.

```
 "agentPoolProfiles": [
      {
        "name": "windowspool2",
        "count": 2,
        "vmSize": "Standard_D2_v2",
        "osType": "Windows",
        "availabilityProfile": "AvailabilitySet",
        "storageProfile": "StorageAccount",
        "distro": "ubuntu"
}
```

should become: 

```
 "agentPoolProfiles": [
      {
        "name": "windowspool2",
        "count": 2,
        "vmSize": "Standard_D2_v2",
        "osType": "Windows",
        "availabilityProfile": "AvailabilitySet",
        "storageProfile": "StorageAccount",
        "distro": ""
}
```

### PR DESCRIPTION
**Pull request by urld**
**What this PR does / why we need it**:
see #2619

**Which issue this PR fixes** : fixes #2619

**Special notes for your reviewer**:

**If applicable**:
- [ ] documentation
- [ ] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:
```NONE
```

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
@urld thank you for your contribution! We had to change the distro logic this morning in #3951 to fix A Swarm issue so you now have a merge conflict unfortunately. Let me know when you were able to fix it and this is ready for review.

**Comment by urld** [comment]
@CecileRobertMichon merge conflict resolved.

File: pkg/acsengine/defaults.go [hunk 1]
```diff
@@ -654,7 +654,7 @@ func setAgentProfileDefaults(a *api.Properties, isUpgrade, isScale bool) {
 				} else {
 					profile.Distro = api.AKS
 				}
-			} else if !a.OrchestratorProfile.IsOpenShift() {
+			} else if !a.OrchestratorProfile.IsOpenShift() && profile.OSType != api.Windows {
```
[the remark this answers is not in the document]
Comment (urld) [reply pkg/acsengine/defaults.go]: oops :/ badly resolved that merge conflict.
done.

**Comment by CecileRobertMichon** [comment]
Thank you for your contribution @urld ! Please make sure you have filled this form out, if you haven't already :)
https://microsoft.qualtrics.com/jfe/form/SV_1NrfDyLTNVFun4x

**Pull request merged**